nilearn/plotting/surf_plotting.py

```diff
@@ -29,13 +31,571 @@
+                      colorbar=False, threshold=None, vmin=None,
+                      vmax=None, title=None, font_size=15,
+                      output_file=None):
+    """Helper function for plot_surf.
+
+    .. versionadded:: 0.8.1
+
+    This function handles surface plotting when the selected
+    engine is plotly.
+
+    .. note::
+        This function assumes that plotly and kaleido are
+        installed.
+
+    """
+    try:
+        import plotly.graph_objects as go
+        import kaleido  # noqa: F401
+    except ImportError:
+        raise ImportError("Using engine='plotly' requires that "
+                          "plotly and kaleido are installed.")
+    x, y, z = coords.T
+    i, j, k = faces.T
+    if cmap is None:
+        cmap = cold_hot
+    bg_data = None
+    if bg_map is not None:
+        bg_data = load_surf_data(bg_map)
+        if bg_data.shape[0] != coords.shape[0]:
+            raise ValueError('The bg_map does not have the same number '
+                             'of vertices as the mesh.')
+    if surf_map is not None:
+        _check_surf_map(surf_map, coords.shape[0])
+        vmin, vmax = _get_bounds(
+            surf_map, vmin, vmax, threshold,
+            symmetric_cmap=False,
+            enforce_symmetric_cmap=False
+        )
+        intensity, colorscale = _get_intensity_and_colorscale_plotly(
+            surf_map, vmin, vmax, cmap, bg_data, threshold
+        )
+    else:
+        if bg_data is None:
+            bg_data = np.ones(coords.shape[0])
+        vmin, vmax = _get_asymmetric_bounds(bg_data)
+        intensity, colorscale = _get_intensity_and_colorscale_plotly(
+            bg_data, vmin, vmax, 'Greys'
+        )
+    mesh_3d = go.Mesh3d(x=x, y=y, z=z, i=i, j=j, k=k,
+                        intensity=intensity,
+                        colorscale=colorscale,
+                        showscale=colorbar,
+                        cmin=vmin, cmax=vmax)
+    cameras_view = _set_view_plot_surf_plotly(hemi, view)
+    fig = go.Figure(data=[mesh_3d])
+    fig.update_layout(scene_camera=CAMERAS[cameras_view],
+                      title=_configure_title_plotly(title, font_size),
+                      **LAYOUT)
+    if output_file is not None:
+        fig.write_image(output_file)
+    return fig
+
+
+def _set_view_plot_surf_matplotlib(hemi, view):
+    """Helper function for plot_surf with matplotlib engine.
+
+    This function checks the selected hemisphere and view, and
+    returns elev and azim.
+    """
+    if hemi not in VALID_HEMISPHERES:
+        raise ValueError(f"hemi must be one of {VALID_HEMISPHERES}")
+    if view not in MATPLOTLIB_VIEWS[hemi]:
+        raise ValueError(f"view must be one of {VALID_VIEWS}")
+    return MATPLOTLIB_VIEWS[hemi][view]
+
+
+def _check_surf_map(surf_map, n_vertices):
+    """Helper function for plot_surf.
+
+    This function checks the dimensions of provided surf_map.
+    """
+    surf_map_data = load_surf_data(surf_map)
```

Comment:
I guess this flattens surf_map when shape is (n_vertices, 1), right ?

Reply:
Yes indeed (I just checked what `nilearn.surface.load_surf_data` does, and it eventually calls `np.squeeze()` on the returned array).

Reply:
Yes indeed (I just checked what `nilearn.surface.load_surf_data` does, and it eventually calls `np.squeeze()` on the returned array).